gitopscli/commands/sync_apps.py

```
@@ -50,18 +49,65 @@ def __sync_apps(team_config_git_repo: GitRepo, root_config_git_repo: GitRepo, gi
+    )
     __commit_and_push(team_config_git_repo, root_config_git_repo, git_user, git_email, apps_config_file_name)
 
 
+def __clean_yaml(values: Dict[str, Any], whitelist: Dict[str, Any], app_name: str) -> Any:
+
+    whitelisted_yml = {k:v for k,v in values.items() if k in whitelist.keys()}
+    missing = set(values).difference(whitelisted_yml)
+    logging.info("Keys removed from %s custom_values.yaml %s", app_name, missing)
+    return whitelisted_yml
+
+
+def __clean_repo_app(root_config_git_repo: GitRepo, team_config_git_repo: GitRepo, app_name: str) -> Any:
+    app_spec_file = team_config_git_repo.get_full_file_path(f"{app_name}/custom_values.yaml")
+    app_spec_whitelist_file = root_config_git_repo.get_full_file_path(f"whitelist.yaml")
+    try:
+        app_config_content = yaml_file_load(app_spec_file) 
+    except FileNotFoundError as ex:
+        logging.warning("no specific app settings file found for %s", app_name)
+        return {}
+    except YAMLException as yex:
+        logging.error("Unable to load %s/custom_values.yaml from app repository, please validate if this is a correct YAML file" , exc_info=yex)
+        return {}
+    #TODO: should sync fail or skip adding custom values (return {})
+    #TODO: which errors should be raised as GitOpsException
+    try: 
+        whitelist_config_content = yaml_file_load(app_spec_whitelist_file) 
+    except FileNotFoundError as ex:
+        logging.warning("no whitelist app settings file found in root repo, by default allowing teamcode only")
+        whitelist_yaml_string = """\
+            teamcode: null
+        """
+        whitelist_config_content = yaml_load(whitelist_yaml_string)
+    except YAMLException as yex:
+        logging.error("Unable to load whitelist.yaml from root repository, please validate if this is a correct YAML file" , exc_info=yex)
+        return {}
+    #TODO: should sync fail, assume default whitelist or skip adding custom values (return {})
```

Taking into account that the sync-apps command is usually run by some pipeline not directly maintained by the individual teams, it might be a bit hard to figure out what went wrong (or even notice that it failed in the first place).

A compromise could be to create a YAML comment in the resulting root config file with the failure message:

```yaml
applications:
  app-xy-staging:
    teamcode: team-xy
   # sync-apps error: 'keydisallowed' not allowed (custom_values.yaml, line 3)
```
This way you neither have to swallow the error nor break the sync.